transient/cli.py

```diff
@@ -281,6 +284,24 @@ def _find_requested_images(
     return images
 
 
+@click.help_option("-h", "--help")
+@click.option("-name", help="Creates a config file with the given name")
+@cli_entry.command("init")
+def init_impl(**kwargs: Any) -> None:
+    """Creates a default Transient config file """
+    args = argparse.Namespace(**kwargs)
+
+    if args.name:
+        config_file = args.name
+    else:
+        config_file = "transient-config.toml"
+
+    with open(config_file, "w") as file:
+        file.write(_config.default_config_file)
+
+    sys.exit(0)
```

Comment:
I think I'm opposed to this 'init' functionality. In my mind, `init` is typically associated with a command that has some special effect on the given directory. `git init` make the directory a git repo, `vagrant init` makes it a vagrant dir, `cargo init`, etc. But transient, by design, has no special effect on the directory. There are no hidden state directories there.

We could just change the name, but then we also have the problem of the default config not having any meaningful default image. The usage of `centos/7:2004.01` seems arbitrary. I think I'd rather just move the 'default' config in to examples in the docs instead.

Reply:
I shameless stole the concept from Vagrant - `vagrant init` initializes the current directory with a Vagrantfile.

It is true that Transient doesn't affect the directory in a special manner.  I'm open to changing the name - the intent of this function is to provide a default configuration file that worked out-of-the-box which developers could easily create and modify:
```
transient init
transient run -config transient-config.toml
```

I would like to push back on keeping a config file example only in docs. During testing, I intend for Transient to create a config and use it, which exercises the config file. Hopefully, this will prevent the config file schema from falling out of date - especially compared to a docs page. However, it would be beneficial to have a single docs page that enumerates all possible config options with examples.

The image I picked was arbitrary.

How would you like to proceed?

Reply:
I agree that I don't want us to get into a position where the docs conflict with the actual schema. I would rather put example configs in a directory and include those files in the docs and the tests. This ensures that _every_ example we give does actually work.

I don't think I am really persuaded by the out-of-the-box concept. I expect a developer to go more in the direction of: 

1. I need to run an ubuntu vm: `transient run -image ubuntu:whaterver -ssh -- -m 1G`
2. Oh, I also need a shared folder `.. -s host:guest`
3. Ah, I guess I also do want to wait for shutdown `... -shutdown-timeout 0`
4. I guess I'll make a config file rather than this long commandline 

Any options we pick for the default file will be arbitrary and not necessarily correct anyway. The usage of `-enable-kvm`, for example, won't work when `/dev/kvm` is not available (e.g., under docker). So, I remain opposed to having `init` functionality like this.

Reply:
That's a well-reasoned argument and I am inclined to agree with you regarding a configuration file driven by need rather than a boilerplate config - especially given the simple interface.

To resolve this, I will:

1. remove the `init` command
2. move the exhaustive config file string to `test/config-files/exhaustive-config`

Let me know if you disagree and/or have any additional comments.

Reply:
Resolved with my proposed solution.